src/biogeophys/CanopyHydrologyMod.F90

```diff
@@ -301,6 +303,23 @@ subroutine CanopyInterceptionAndThroughfall(bounds, &
      ! Update snocan and liqcan based on interception, for bulk water and each tracer
      do i = water_inst%bulk_and_tracers_beg, water_inst%bulk_and_tracers_end
         associate(w => water_inst%bulk_and_tracers(i))
+        ! Get rid of tiny snocan (tracer and bulk)
+        ! Very small snocan (< rel_epsilon**2) will be set to zero
+        ! See NumericsMod for rel_epsilon value
+        snocan_baseline(bounds%begp:bounds%endp) = rel_epsilon  ! this gets multiplied by rel_epsilon in truncate_small_values
+        call truncate_small_values(num_soilp, filter_soilp, begp, endp, &
+           snocan_baseline(begp:endp), b_waterstate_inst%snocan_patch(begp:endp))
+        ! Explanation for why we use this do-loop instead of call
+        ! truncate_small_values for w%waterstate_inst%snocan_patch:
+        ! call truncate_small_values would set w%waterstate_inst%snocan_patch
+        ! to zero only when w%waterstate_inst%snocan_patch < rel_epsilon**2
+        ! while it needs to be zero every time that
+        ! b_waterstate_inst%snocan_patch is zero
+        do p = begp, endp
+           if (b_waterstate_inst%snocan_patch(p) == 0._r8) then
+              w%waterstate_inst%snocan_patch(p) = b_waterstate_inst%snocan_patch(p)
```

Comment:
Oh, probably better to replace the if-statement with
`w%waterstate_inst%snocan_patch(p) = max(0._r8, b_waterstate_inst%snocan_patch(p))`

Or not?

Reply:
...actually, no, because the left-hand-side (tracer) may need to equal bulk * ratio.